packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1051,7 +1066,8 @@ class TerminalHandler {
       await _commonTerminalInputHandler(command);
     } catch (error, st) {
       printError('$error\n$st');
-      await _cleanUpAndExit(null);
+      await _cleanUp(null);
+      rethrow;
```

Comment:
What will catch this? Why not also a throwToolExit?

Reply:
So previously we caught everything and then exited immediately before error reporting. Now we're only catching RpcException and throwing a tool exit in the TerminalHandler.

This rethrow allows us to both rethrow the existing tool exit and handle cleanups so we exit nicely. If there are other exceptions that we're not handling, we should probably let those bubble up so we can fix them.